pkg/gadget-service/api-helpers/aphelpers_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package apihelpers provides some helper functions for the API package; these were extracted into this package
+// to avoid having additional dependencies on the API package itself
+package apihelpers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetStringValuesPerDataSource(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expected    map[string]string
+		expectedErr bool
+	}{
+		{
+			name:     "empty input",
+			input:    "",
+			expected: map[string]string{},
+		},
+		{
+			name:     "valid without datasource",
+			input:    "50",
+			expected: map[string]string{"": "50"},
+		},
+		{
+			name:     "valid with datasource",
+			input:    "datasource1:10",
+			expected: map[string]string{"datasource1": "10"},
+		},
+		{
+			name:     "valid with multiple datasources",
+			input:    "datasource1:10,datasource2:20",
+			expected: map[string]string{"datasource1": "10", "datasource2": "20"},
+		},
+		{
+			name:     "valid with empty element",
```

Comment:
I'm not sure whether this should really be valid - but we can discuss on it later.